structure-macro-impl/src/lib.rs

```diff
@@ -2,9 +2,25 @@
 
 extern crate proc_macro;
 
+#[cfg(not(feature = "std"))]
+extern crate core;
+#[cfg(not(feature = "std"))]
```

Comment:
`no_std` usually means "no alloc as well". Otherwise the usability of the crate is much smaller again.

Reply:
OK, well, we do need `alloc` for `alloc::string::String`. See comment and log below.